### PR TITLE
kenel, testcase: Fix wrong usages in testcases and Add missing logics for waitid

### PR DESCRIPTION
**Pull request by jeongarmy**
- testcase/kernel: Fix wrong condition for checking result in testcase for waitid with P_ALL
- testcase/kernel: Fix wrong idtype for waitpid with P_PID
- kernel/sched: Add missing logic for verifying idtype

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [7dca1298d172db06cffe67fe3869e80c796940d1] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [7dca1298d172db06cffe67fe3869e80c796940d1] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [df2d2ca94306b3bd495b160d7321577d92c4cc6f] -

**Comment by seinfra** [comment]
Target : [d6a9423514df042a5060c3230add38d8019f6cc7] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [d6a9423514df042a5060c3230add38d8019f6cc7] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e1b42118e283fe1674cd2eead4636f55cd62ae42] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [e1b42118e283fe1674cd2eead4636f55cd62ae42] - Code Rule Check (C++) OK.

File: os/kernel/sched/sched_waitid.c [hunk 1]
```diff
@@ -180,26 +180,25 @@ int waitid(idtype_t idtype, id_t id, FAR siginfo_t *info, int options)
...
+	 * We support P_ALL and P_PID ID types only.
+	 * And none of the options are supported except for WEXITED (which must be
 	 * provided.  Currently SIGCHILD always reports CLD_EXITED so we cannot
 	 * distinguish any other events.
 	 */
-
-#ifdef CONFIG_DEBUG
-	if (options != WEXITED) {
+	if (((idtype != P_ALL) && (idtype != P_PID)) || (options != WEXITED)) {
 		set_errno(ENOSYS);
```
Comment (jeongchanKim) [inline os/kernel/sched/sched_waitid.c]: Is it correct? I think EINVAL should be set to errno.
In posix documentation,
EINVAL  returns when an invalid value was specified for options, or idtype and id specify an invalid set of processes.

**Pull request merged**